eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/configuration/MongoClientCreator.java

```diff
@@ -0,0 +1,39 @@
+package uk.ac.ebi.eva.accession.core.configuration;
+
+import com.mongodb.*;
+import org.springframework.beans.factory.ObjectProvider;
+import org.springframework.boot.autoconfigure.mongo.MongoClientFactory;
+import org.springframework.boot.autoconfigure.mongo.MongoProperties;
+import org.springframework.core.env.Environment;
+import uk.ac.ebi.eva.commons.mongodb.utils.MongoUtils;
+
+import java.io.UnsupportedEncodingException;
+import java.util.Objects;
+
+public class MongoClientCreator {
+    public static MongoClient getMongoClient(MongoProperties properties, ObjectProvider<MongoClientOptions> options,
+                                             Environment environment, String readPreference)
+            throws UnsupportedEncodingException {
+        MongoClientOptions mongoClientOptions = options.getIfAvailable();
+        // Only set the URI if it isn't already set
+        if (Objects.isNull(properties.getUri())) {
+            // Weirdly, MongoClient instantiation works without authentication mechanism
+            // in the eva-accession project but does not work in the eva-pipeline project
+            // So we explicitly pass it as null here
+            properties.setUri(MongoUtils.constructMongoClientURI(properties.getHost(), properties.getPort(),
+                    properties.getDatabase(), properties.getUsername(), (Objects.nonNull(properties.getPassword()) ?
+                            String.valueOf(properties.getPassword()) : ""),
+                    properties.getAuthenticationDatabase(), null, readPreference).getURI());
+        }
+        properties.setHost(null);
```

Comment:
When there are both the URI and the host parameters, Spring gets confused and throws some weird error messages because the former already has the hostnames embedded in it.